### PR TITLE
Restore `prove` into `metal` feature

### PR DESCRIPTION
**Pull request by capossele**
With #2092 dropping the `metal` feature and #2179 restoring it as an no-op feature, this PR restores the `prove` feature into it, with the goal of maintaining semver compatibility.

### CODE REVIEW

File: risc0/zkvm/Cargo.toml [hunk 1]
```diff
@@ -136,7 +136,7 @@ dual = []
 # The zkVM exposes a getrandom implementation that panics by default. This will
 # expose a getrandom implementation that uses the `sys_random` ecall.
 getrandom = ["risc0-zkvm-platform/getrandom"]
-metal = []
+metal = ["prove"]
```
Comment (flaub) [inline risc0/zkvm/Cargo.toml:139]:
```suggestion
metal = ["prove", "risc0-zkp/metal"]
```

File: risc0/zkp/Cargo.toml [hunk 2]
```diff
@@ -58,7 +58,7 @@ features = ["prove", "std"]
 circuit_debug = []
 cuda = ["dep:cust", "prove", "risc0-sys/cuda"]
 default = []
-metal = []
+metal = ["prove"]
```
Comment (flaub) [inline risc0/zkp/Cargo.toml:61]:
```suggestion
metal = ["prove", "risc0-sys/metal"]
```

Comment (capossele) [reply risc0/zkp/Cargo.toml:61]: I haven't originally included `risc0-sys/metal` as it is an empty feature.

File: risc0/zkvm/Cargo.toml (hunk 1 above)
Comment (capossele) [reply risc0/zkvm/Cargo.toml:139]: Also here, I was under the assumption that by including `prove` we get 
```
"risc0-circuit-recursion/prove",
 "risc0-circuit-rv32im/prove",
 "risc0-zkp/prove",
 ```
 Wouldn't that be enough?

**Review by flaub — APPROVED** [review]
I haven't been able to test this. How can we test this?

**Comment by capossele** [comment]
I've only checked that by invoking the `metal` feature on any test gated behind `prove` works, e.g.,: 
`cargo test -F metal --package risc0-zkvm --lib -- receipt_claim::tests::merge_receipt_claim --exact --show-output`

**Comment by capossele** [comment]
thanks to @SchmErik for suggesting to add a specific test for it

**Pull request merged**